subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -216,6 +225,55 @@ static int hawkbit_settings_set(const char *name, size_t len, settings_read_cb r
 		return rc;
 	}
 
+	size_t next_len = settings_name_next(name, &next);
```

Comment:
I think it probably makes sense to use just one level? (i.e. `hawkbit/server_addr`, `hawkbit/server_port`, `hawkbit/action_id`) then you will be able to literally copy paste the same thing over and over again, be it `settings_name_steq` or `strncmp(..)`

seems like we can factorize this function quite abit